From e8fa0f8f97cfb3fa0c717d5ab37d4ab371ee1c50 Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Tue, 15 Jan 2019 17:18:51 +0800
Subject: [PATCH] Fix disallow user switch for CTS Device owner Test

if devices are managed by admin, the current user doesn't allow
to manage the user.

bug: 120237874
Test: Launch CTS-VERIFIER, follow the instruction to start the
      Device owner Test ->Disallow user switch

Change-Id: I0cf8d1e9f701cc49ff7e400e325294480803422d
Tracked-On: https://jira.devtools.intel.com/browse/OAM-71178
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
Signed-off-by: Yong Yao <yong.yao@intel.com>
Reviewed-on: https://android.intel.com:443/642677
---
 .../quicksettings/QuickSettingFragment.java         | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/com/android/car/settings/quicksettings/QuickSettingFragment.java b/src/com/android/car/settings/quicksettings/QuickSettingFragment.java
index 92d3d9f..aaddef8 100644
--- a/src/com/android/car/settings/quicksettings/QuickSettingFragment.java
+++ b/src/com/android/car/settings/quicksettings/QuickSettingFragment.java
@@ -19,6 +19,7 @@ import android.car.drivingstate.CarUxRestrictions;
 import android.car.user.CarUserManagerHelper;
 import android.content.pm.UserInfo;
 import android.os.Bundle;
+import android.os.UserManager;
 import android.view.View;
 import android.view.View.OnClickListener;
 import android.widget.ImageView;
@@ -97,6 +98,7 @@ public class QuickSettingFragment extends BaseFragment {
                 .addTile(new DayNightTile(getContext(), mGridAdapter))
                 .addTile(new CelluarTile(getContext(), mGridAdapter))
                 .addSeekbarTile(new BrightnessTile(getContext()));
+        mUserSwitcherBtn.setVisibility(showUserSwitcher() ? View.VISIBLE : View.INVISIBLE);
     }
 
     @Override
@@ -115,6 +117,17 @@ public class QuickSettingFragment extends BaseFragment {
         userSwitcherText.setText(currentUserInfo.name);
     }
 
+    private boolean showUserSwitcher() {
+        UserManager userManager = UserManager.get(getContext());
+        if (userManager.isDeviceInDemoMode(getContext()) || !userManager.supportsMultipleUsers()) {
+            return false;
+        }
+        if (userManager.hasUserRestriction(UserManager.DISALLOW_USER_SWITCH)) {
+            return false;
+        }
+        return true;
+    }
+
     /**
      * Quick setting fragment is distraction optimized, so is allowed at all times.
      */
-- 
2.20.0

